model/userdata.go

```diff
@@ -147,3 +148,71 @@ func validateCoreosCloudInit(content []byte) error {
 func validateNone([]byte) error {
 	return nil
 }
+
+func (self UserDataPart) Base64IgnitionConfigRef() (string, error) {
+	base64Part, err := self.Base64(false)
+	if err != nil {
+		return "", fmt.Errorf("failed to generate ignition config: %v", err)
+	}
+
+	// TODO Append user-provided ignition config via `ignition.config.append.source`
+	// once it supports sourcing from a s3 object
+	// https://coreos.com/ignition/docs/0.14.0/configuration-v2_0.html
+
+	head, err := json.Marshal(`{
+  "ignition": {
+    "version": "2.0.0"
```

Comment:
In the future, to allow customizing the ignition config on user-side i.e. retain the flexibility of having all configs in a text form, we can add `ignition.config.append.source` which points to the user-provided ignition config json uploaded to s3.